### ISSUE TITLE
LC0048 - false positive with this.globalLabel

### ISSUE THREAD

**Issue by ChristianHovenbitzer** [StefanMaron/BusinessCentral.LinterCop#900]
An `Error(this.GlobalLabel)` triggers this rule. Probably unintended

![Image](https://github.com/user-attachments/assets/03e53d1f-d3a6-4e0a-a4e2-2eaa771224f9)

```
codeunit 51021 MyCodeunit
{

    local procedure Test()
    begin
        Error(this.GlobalLable);
    end;

    var
        GlobalLable: Label 'Global Lable';
}
```

**Comment by Arthurvdv** [comment]
Thanks! Some rules need some rework for handling the new this keyword, as it changes to type of parameter in the code analyzer.

I believe I've resolved this in the Pre-release [v0.33.1](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.33.1) version of the LinterCop. Could you verify if it now works as expected?

**Comment by ChristianHovenbitzer** [comment]
Works with the prerelease. I should check that more often before opening issues. Thanks

**Comment by Arthurvdv** [comment]
Great to hear it's now working!

I see my comment wasn't clear, I created an updated version of the pre-release thanks to your reported issue, where it wasn't already solved before. Thanks again for reporting this, much appreciated 🤗
